_integration-tests/tests/civisibility/main_test.go

```diff
@@ -0,0 +1,159 @@
+	events.
+		CheckEventsByType("test_session_end", 1)
+
+	// module event
+	events.
+		CheckEventsByType("test_module_end", 1).
+		CheckEventsByResourceName("orchestrion/integration/tests/civisibility", 1)
+
+	// test suite event
+	events.CheckEventsByType("test_suite_end", 1).
+		CheckEventsByResourceName("testing_test.go", 1)
+
+	// test events
+	testEvents := events.CheckEventsByType("test", 10)
+	normalTests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestNormal", 1).
+		CheckEventsByTagAndValue("test.status", "pass", 1)
+	failTests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestFail", 1).
+		CheckEventsByTagAndValue("test.status", "fail", 1).
+		CheckEventsByTagAndValue("error.type", "Fail", 1).
+		CheckEventsByTagAndValue("error.message", "failed test", 1)
+	errorTests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestError", 1).
+		CheckEventsByTagAndValue("test.status", "fail", 1).
+		CheckEventsByTagAndValue("error.type", "Error", 1).
+		CheckEventsByTagAndValue("error.message", "My error test", 1)
+	errorFTests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestErrorf", 1).
+		CheckEventsByTagAndValue("test.status", "fail", 1).
+		CheckEventsByTagAndValue("error.type", "Errorf", 1).
+		CheckEventsByTagAndValue("error.message", "My error test: TestErrorf", 1)
+	skipTests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestSkip", 1).
+		CheckEventsByTagAndValue("test.status", "skip", 1).
+		CheckEventsByTagAndValue("test.skip_reason", "My skipped test", 1)
+	skipfTests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestSkipf", 1).
+		CheckEventsByTagAndValue("test.status", "skip", 1).
+		CheckEventsByTagAndValue("test.skip_reason", "My skipped test: TestSkipf", 1)
+	skipNowTests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestSkipNow", 1).
+		CheckEventsByTagAndValue("test.status", "skip", 1)
+	testWithSubtests := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestWithSubTests", 1).
+		CheckEventsByTagAndValue("test.status", "pass", 1)
+	testWithSubtestsChild1 := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestWithSubTests/Sub1", 1).
+		CheckEventsByTagAndValue("test.status", "pass", 1)
+	testWithSubtestsChild2 := testEvents.
+		CheckEventsByResourceName("testing_test.go.TestWithSubTests/Sub2", 1).
+		CheckEventsByTagAndValue("test.status", "pass", 1)
+
+	// remaining must be 0
+	testEvents.
+		Except(
+			normalTests,
+			failTests,
+			errorTests,
+			errorFTests,
+			skipTests,
+			skipfTests,
+			skipNowTests,
+			testWithSubtests,
+			testWithSubtestsChild1,
+			testWithSubtestsChild2).
+		HasCount(0)
+
+	// All previous checks will cause panic if they fail so we can safely exit with 0 here
+	os.Exit(0)
+}
+
+func enableCiVisibilityEndpointMock() *httptest.Server {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/api/v2/citestcycle" {
```

Comment:
This condition should be flipped, so the "happy case" is un-indented...

Really what this function is:

```suggestion
		if r.URL.Path != "/api/v2/citestcycle" {
			w.WriteHeader(http.StatusBadRequest)
			return
		}
```

I also wonder if that should be an HTTP 404 here; but it's not really important...

Reply:
done in [92ee768](https://github.com/DataDog/orchestrion/pull/340/commits/92ee768360c07a61c2311485193b2236c93be0b1)